botcity/datapool/datapool.py

```diff
@@ -0,0 +1,180 @@
+            req.raise_for_status()
+
+    def deactivate(self):
+        url = f'{self.maestro.server}/api/v2/datapool/{self.label}'
+        data = self.to_dict()
+        data['active'] = False
+        with requests.post(url, data=json.dumps(data), headers=self.maestro._impl._headers(),
+                           timeout=self.maestro.timeout) as req:
+            if req.ok:
+                self._update_from_json(payload=req.content)
+                return True
+            req.raise_for_status()
+
+    def is_activated(self):
+        url = f'{self.maestro.server}/api/v2/datapool/{self.label}'
+
+        with requests.get(url, headers=self.maestro._impl._headers(), timeout=self.maestro.timeout) as req:
+            if req.ok:
+                self._update_from_json(payload=req.content)
+                return self.active
+            req.raise_for_status()
+
+    def summary(self) -> dict:
+        url = f'{self.maestro.server}/api/v2/datapool/{self.label}/summary'
+
+        with requests.get(url, headers=self.maestro._impl._headers(), timeout=self.maestro.timeout) as req:
+            if req.ok:
+                self._update_count(payload=req.content)
+                return json.loads(req.content)
+            req.raise_for_status()
+
+    def create_entry(self, entry: DataPoolEntry):
+        url = f'{self.maestro.server}/api/v2/datapool/{self.label}/push'
+
+        with requests.post(url, data=entry.to_json(), headers=self.maestro._impl._headers(),
+                           timeout=self.maestro.timeout) as req:
+            if req.ok:
+                entry.update_from_json(payload=req.content)
+                return json.loads(req.content)
+            req.raise_for_status()
+
+    def is_empty(self):
+        summary = self.summary()
+        if summary.get("countPending", 0) == 0:
+            return True
+        return False
+
+    def save_entry(self, entry: DataPoolEntry):
+        url = f'{self.maestro.server}/api/v2/datapool/{self.label}/entry/{entry.entry_id}'
+        data = entry.json_to_update()
+        with requests.post(url, data=data, headers=self.maestro._impl._headers(),
+                           timeout=self.maestro.timeout) as req:
+            if req.ok:
+                entry.update_from_json(payload=req.content)
+                return json.loads(req.content)
+            req.raise_for_status()
+
+    def next(self, task_id: int) -> DataPoolEntry:
+        url = f'{self.maestro.server}/api/v2/datapool/{self.label}/pull'
+        with requests.get(url, headers=self.maestro._impl._headers(), timeout=self.maestro.timeout) as req:
+            if req.ok:
+                entry = DataPoolEntry(task_id=task_id)
+                entry.maestro = self.maestro
+                entry.update_from_json(payload=req.content)
+                entry.state = StateEnum.PROCESSING
+                try:
+                    entry = entry.save()
+                except Exception as error:
+                    print(error)
```

Comment:
When we fetch an entry, it is the DataPool job to set it as processing, not the client library.